### PR TITLE
qt5.qtdatavis3d: init module

### PR DESCRIPTION
**Pull request by panicgh**
###### Description of changes

New module.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>libsForQt5.qt5.qtdatavis3d</li>
    <li>libsForQt5.qt5.qtdatavis3d.bin</li>
    <li>libsForQt5.qt5.qtdatavis3d.dev</li>
  </ul>
</details>

### CODE REVIEW

File: pkgs/development/libraries/qt-5/5.15/default.nix [hunk 1]
```diff
@@ -202,11 +203,11 @@ let
 
       env = callPackage ../qt-env.nix {};
       full = env "qt-full-${qtbase.version}" ([
-        qt3d qtcharts qtconnectivity qtdeclarative qtdoc qtgraphicaleffects
-        qtimageformats qtlocation qtmultimedia qtquickcontrols qtquickcontrols2
-        qtscript qtsensors qtserialport qtsvg qttools qttranslations
-        qtvirtualkeyboard qtwebchannel qtwebengine qtwebkit qtwebsockets
-        qtwebview qtx11extras qtxmlpatterns qtlottie
+        qt3d qtcharts qtconnectivity qtdatavis3d qtdeclarative qtdoc
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/qt-5/5.15/default.nix:206]: Not sure if that should be part of full or not. Also I wouldn't reflow here to reduce the diff.

Comment (panicgh) [reply pkgs/development/libraries/qt-5/5.15/default.nix]: I can't judge about "full or not", but I undid the reflow to reduce the diff.

**Comment by K900** [comment]
Is this actually used by anything?

**Comment by panicgh** [comment]
I locally packaged (and plan to upstream) [nitrokey-app2](https://github.com/Nitrokey/nitrokey-app2/) (the new rewrite in python). For this, I had to package PyQt5-stubs which has a ton of dependencies, among which is python3Packages.pyqtdatavisualization, which needs qt5.qtdatavis3d.

Instead of creating a MR for qt5.qtdatavis3d, python3Packages.{pyqtchart, pyqt3d, pyqtdatavisualization}, fixes to pyqt5 (fix QtConnectivity, add QtSerialPort, add QtDesigner), PyQt5-stubs, nitrokey-app2, I chose to create separate PRs. My plan would be this PR (qt5.qtdatavis3d), then another one for all the pyqt changes, then nitrokey-app2.

**Comment by K900** [comment]
All right, just making sure.

**Pull request merged**